src/Compilers/CSharp/Portable/BoundTree/OutDeconstructVarPendingInference.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal partial class OutDeconstructVarPendingInference
+    {
+        public BoundDeconstructValuePlaceholder Placeholder;
+
+        public BoundDeconstructValuePlaceholder SetInferredType(TypeSymbol type, bool success)
+        {
+            Placeholder = new BoundDeconstructValuePlaceholder(this.Syntax, type, hasErrors: this.HasErrors || !success);
```

Comment:
I assume Placeholder is expected to be set exactly once. We should assert that Placeholder is null before assigning it.